core/server/master/src/main/java/alluxio/master/file/meta/MutableInodeDirectory.java

```diff
@@ -200,6 +202,9 @@ public static MutableInodeDirectory fromJournalEntry(InodeDirectoryEntry entry)
     } else {
       ret.mDefaultAcl = new DefaultAccessControlList();
     }
+    if (entry.getXAttrCount() > 0) {
+      ret.setXAttr(entry.getXAttrMap());
```

Comment:
Do we not need to update `toJournalEntry` and `toProto`, like the file counterpart?

Reply:
I may have missed that...

Reply:
good catch!